modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDuplicatedHandler.es.js

```diff
@@ -140,7 +127,9 @@ const handleFieldDuplicated = (props, state, event) => {
 	const {fieldName} = event;
 	const {pages} = state;
 
-	const originalField = FormSupport.findFieldByName(pages, fieldName);
+	const originalField = JSON.parse(
+		JSON.stringify(FormSupport.findFieldByName(pages, fieldName))
+	);
```

Comment:
@carolmariaabb, why this is necessary?

Reply:
If you pass the variable _originalField_ by reference (without using JSON.parse(JSON.stringify())),  [fieldDuplicatedHandler.es.js#L29-L34](https://github.com/carolmariaabb/liferay-portal/blob/91ac14b75b10335551e4488654f74fb1223880b6/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDuplicatedHandler.es.js#L29-L34) will change the validation parameter for the _originalField_ and the expression will become `isUrl(CopyOfText)` instead of `isUrl(Text)`, for example.

Reply:
@carolmariaabb, can you please give me more details about it? From what I can see, this change is only converting the return of `FormSupport.findFieldByName` to a String and then converts the String to a JSON.

Reply:
I made this solution based on [this link](https://stackoverflow.com/questions/18359093/how-to-copy-javascript-object-to-new-variable-not-by-reference). 

You can find more details also in [here](https://codeburst.io/explaining-value-vs-reference-in-javascript-647a975e12a0) (**Pure Function**s topic).

> We use JSON.stringify to transform the object we’re passed into a string, and then parse it back into an object with JSON.parse. By performing this transformation and storing the result in a new variable, we’ve created a new object... The new object has the same properties as the original but it is a distinctly separate object in memory.

Reply:
Thanks for your detailed explanation, @carolmariaabb. 🙂